design-documents/graph-ql/coverage/search.md

```diff
@@ -0,0 +1,79 @@
+# Queries
+
+```graphql
+type Query {
+    multiSearch(phrase: String!, productSize: Int = 10): MultiSearchResponse!
+    #Filter supports multiple clauses which will be wrapped in logical AND operator
+    productSearch(phrase: String!, filter: [SearchClauseInput], sort: [ProductSearchSortInput]): ProductSearchResponse!
+}
+
+input ProductSearchSortInput
+{
+    attribute: String!
+    direction: SortEnum!
+}
+
+# If from or to fields are omitted, $gte or $lte filter will be applied
+input SearchRangeInput {
+    from: Int
+    to: Int
+}
+
+input SearchClauseInput {
+    attribute: String!
+    in: [String]
+    eq: String
+    range: SearchRangeInput
+}
+
+type ProductSearchResponse {
+    items: [ProductSearchItem]
+    hits: Int!
+    facets: [Aggregation]
+    facetsValues: [Aggregation]
+}
+
+type MultiSearchResponse {
+    products: ProductSearchResponse
+    suggestions: [String]
+}
+
+type ProductSearchItem {
+    product: ProductInterface!
+    highlights: [Highlight]
+}
+
+type Highlight {
+    attribute: String!
+    value: String!
+    matchedWords: [String]!
+}
+
+interface Bucket {
+    #Human readable bucket title
+    title: String!
+    attribute: String!
+}
+
+type StatsBucket implements Bucket {
+    min: Int!
+    max: Int!
+}
```

Comment:
what would we have the stats bucket for? Should we return stats for the whole query or for each particular bucket? 

If it contains stats I think we should return also median/average/stddev etc.
Like an example here 

```
curl http://localhost:8983/solr/techproducts/query -d '
q=memory&
fq=inStock:true&
json.facet={
  "avg_price" : "avg(price)",
  "num_suppliers" : "unique(manu_exact)",
  "median_weight" : "percentile(weight,50)"
}'

The response to the facet request above will start with documents matching the root domain (docs containing "memory" with inStock:true) then calculate and return the requested metrics:

 "facets" : {
    "count" : 4,
    "avg_price" : 109.9950008392334,
    "num_suppliers" : 3,
    "median_weight" : 352.0
  }
```
https://lucene.apache.org/solr/guide/7_5/json-facet-api.html

Reply:
I think we have only these use cases for now. We can extend stats bucket later.